src/ray/object_manager/test/spilled_object_test.cc

```diff
@@ -217,76 +250,81 @@ TEST(SpilledObjectTest, GetNumChunks) {
                                               ray::rpc::Address(), true /* skip_write */);
   // file corrupted.
-  ASSERT_FALSE(
-      SpilledObject::CreateSpilledObject(object_url1, 2 /* chunk_size */).has_value());
+  ASSERT_FALSE(SpilledObject::CreateSpilledObject(object_url1).has_value());
+}
+
+TEST(MemoryObjectReaderTeset, Getters) {
+  rpc::Address owner_address;
+  owner_address.set_raylet_id("nonsense");
+  std::string data("ata");
+  std::string metadata("metadata");
+  auto obj = CreateMemoryObjectReader(data, metadata, owner_address);
+  ASSERT_EQ(data.size(), obj.GetDataSize());
+  ASSERT_EQ(metadata.size(), obj.GetMetadataSize());
+  ASSERT_EQ(data.size() + metadata.size(), obj.GetObjectSize());
+  ASSERT_EQ(owner_address.raylet_id(), obj.GetOwnerAddress().raylet_id());
 }
 
 namespace {
 void AssertGetChunkWorks(std::string metadata, std::string data,
                          std::vector<uint64_t> chunk_sizes) {
+  rpc::Address owner_address;
+  owner_address.set_raylet_id("nonsense");
+
   std::string expected_output = data + metadata;
-  chunk_sizes.push_back(expected_output.size());
+  if (expected_output.size() != 0) {
+    chunk_sizes.push_back(expected_output.size());
+  }
   auto object_url = CreateSpilledObjectOnTmp(10 /* object_offset */, data, metadata,
                                              ray::rpc::Address());
 
   // check that we can reconstruct the output by concatinating chunks with different
   // chunk_size, and the size of chunk is expected.
   for (auto chunk_size : chunk_sizes) {
-    auto optional_object = SpilledObject::CreateSpilledObject(object_url, chunk_size);
+    auto optional_object = SpilledObject::CreateSpilledObject(object_url);
     ASSERT_TRUE(optional_object.has_value());
-    std::string actual_output_by_chunks;
-    for (uint64_t i = 0; i < optional_object->GetNumChunks(); i++) {
-      auto chunk = optional_object->GetChunk(i);
-      ASSERT_TRUE(chunk.has_value());
-      ASSERT_GE(chunk_size, chunk->size());
-      if (i + 1 != optional_object->GetNumChunks()) {
-        ASSERT_EQ(chunk_size, chunk->size());
+
+    std::vector<ChunkObjectReader> readers{
+        {std::make_shared<SpilledObject>(std::move(optional_object.value())), chunk_size},
+        {std::make_shared<MemoryObjectReader>(
+             CreateMemoryObjectReader(data, metadata, owner_address)),
+         chunk_size}};
+
+    for (auto &reader : readers) {
+      std::string actual_output_by_chunks;
+      for (uint64_t i = 0; i < reader.GetNumChunks(); i++) {
+        auto chunk = reader.GetChunk(i);
+        ASSERT_TRUE(chunk.has_value());
+        ASSERT_GE(chunk_size, chunk->size());
+        if (i + 1 != reader.GetNumChunks()) {
+          ASSERT_EQ(chunk_size, chunk->size());
+        }
+        actual_output_by_chunks.append(chunk.value());
       }
-      actual_output_by_chunks.append(chunk.value());
+      ASSERT_EQ(expected_output, actual_output_by_chunks);
```

Comment:
Can we also test other methods like GetDataSection, GetMetadataSection

Reply:
I had some implicit test for it. maybe it's time to add explicit test for it!